modules/adfs/lib/IdP/ADFS.php

```diff
@@ -96,6 +96,7 @@ public static function ADFS_SignResponse($response, $key, $cert) {
 		$objXMLSecDSig->setCanonicalMethod(XMLSecurityDSig::EXC_C14N);	
 		$responsedom = \SAML2\DOMDocumentFactory::fromString(str_replace ("\r", "", $response));
 		$firstassertionroot = $responsedom->getElementsByTagName('Assertion')->item(0);
+		/* TODO: this hardcodes SHA-1. Can we upgrade to something better? */
```

Comment:
This can be safely set to SHA-256 for ADFS.. The code has significantly changed on master though